main.tf

```diff
@@ -22,6 +24,30 @@ resource "aws_iam_policy" "this" {
       Effect   = "Allow"
       Resource = "*"
       },
+      {
+        "Action" : "sts:AssumeRole",
+        "Effect" : "Allow",
+        "Resource" : [
+          "arn:aws:iam::${data.aws_caller_identity.aws.account_id}:role/grid-s3-access-*"
+        ],
+        "Condition" : {
+          "StringEquals" : {
+            "aws:ResourceAccount" : data.aws_caller_identity.aws.account_id
+          }
+        }
```

Comment:
Is this strictly needed? 

Can you add more detail in the PR description how will this be used, and who is doing it, assuming which role?